typed-racket-lib/typed-racket/types/printer.rkt

```diff
@@ -378,7 +378,10 @@
       [(Value: '()) null]))
   (match type
     ;; if we know how it was written, print that
-    [(? Rep-stx a) (syntax->datum (Rep-stx a))]
+    [(? Rep-stx a)
+     (if (Error? a)
+         `(Error ,(syntax->datum (Rep-stx a)))
+         (syntax->datum (Rep-stx a)))]
```

Comment:
The original code was fine, I'd just have move it a line earlier. Just a nitpick anyway.